File: app/src/main/java/net/mieczkowski/yelpbusinessexample/controllers/base/BaseController.java
package net.mieczkowski.yelpbusinessexample.controllers.base;

import com.bluelinelabs.conductor.Controller;
import com.bluelinelabs.conductor.ControllerChangeHandler;
import com.bluelinelabs.conductor.Router;
import com.bluelinelabs.conductor.RouterTransaction;
import com.bluelinelabs.conductor.changehandler.HorizontalChangeHandler;

import net.mieczkowski.yelpbusinessexample.interfaces.IToolbar;

/**
 * Created by Josh Mieczkowski on 10/19/2017.
 */

public abstract class BaseController extends ButterKnifeController {

    public void show(Controller controller) {
        show(controller, new HorizontalChangeHandler());
    }

    public void show(Controller controller, ControllerChangeHandler controllerChangeHandler) {
        show(controller.getRouter(), controllerChangeHandler);
    }

    public void show(Router router) {
        show(router, new HorizontalChangeHandler());
    }

    public void show(Router router, ControllerChangeHandler controllerChangeHandler) {
        if (!router.popToTag(getClass().getName())) {
            router.pushController(RouterTransaction.with(this)
                    .popChangeHandler(controllerChangeHandler)
                    .pushChangeHandler(controllerChangeHandler)
                    .tag(getClass().getName()));
        }
    }

    protected IToolbar getIToolbar() {
        if (!(getActivity() instanceof IToolbar)) {
            throw new RuntimeException("Need to implement IToolbar");
        }

        return (IToolbar) getActivity();
    }

    public void setTitle(String title) {
        getIToolbar().getSupportActionBar().setTitle(title);
    }

}

File: app/src/main/java/net/mieczkowski/yelpbusinessexample/controllers/base/BaseController.kt
package net.mieczkowski.yelpbusinessexample.controllers.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bluelinelabs.conductor.RestoreViewOnCreateController
import net.mieczkowski.yelpbusinessexample.interfaces.IToolbar
import org.koin.standalone.KoinComponent

/**
 * Created by Josh Mieczkowski on 10/19/2017.
 */

abstract class BaseController(args: Bundle? = null) : RestoreViewOnCreateController(args), KoinComponent {

    protected val iToolbar: IToolbar
        get() {
            if (activity !is IToolbar) {
                throw RuntimeException("Need to implement IToolbar")
            }

            return activity as IToolbar
        }

    protected abstract fun getLayoutID(): Int

    protected abstract fun onViewBound(view: View, savedViewState: Bundle?)

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup, savedViewState: Bundle?): View {
        val view = inflater.inflate(getLayoutID(), container, false)
        onViewBound(view, savedViewState)

        return view
    }

    fun setTitle(title: String?) {
        iToolbar.supportActionBar.title = title
    }

}
